practice/validator/site.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+/**
+ * Tests whether the string matches the regular expression
+ * @param  {string} testString The string to test
+ * @param  {reg exp} regExp    The regular expression.  Not a string.
+ * @return {boolean}           True if there is a match, false otherwise.
+ */
+function stringValidator(testString, regExp) {
+  return regExp.test(testString);
+}
+
+/**
+ * The the string entered on the form.
+ * @param  {jQuery selector} entryItem The selector for a text input.
+ * @return {string}                    The string entered on the form.
+ */
+function getFieldEntry(entryItem) {
+  return entryItem.val();
+}
+
+/**
+ * displayInvalid() highlights the input box and un-hides the warning message.
+ */
+function displayInvalid(entryItem) {
+  entryItem.addClass('invalid');
+  $(entryItem).next().addClass('visible-warning');
```

Comment:
We talked about this in person, but finding a way to more flexibly link the `input` element with its warning box would be a nice enhancement. Either using a structural sub-selection of a parent element, or having an attribute on the `input` that explicitly specifies the ID of the warning box would work.